core/arch/arm/plat-stm32mp1/shared_resources.c

```diff
@@ -596,6 +610,8 @@ static void set_etzpc_secure_configuration(void)
 			    shres2decprot_attr(STM32MP1_SHRES_SPI6));
 	config_lock_decprot(STM32MP1_ETZPC_I2C4_ID,
 			    shres2decprot_attr(STM32MP1_SHRES_I2C4));
+	config_lock_decprot(STM32MP1_ETZPC_I2C5_ID,
+			    shres2decprot_attr(STM32MP1_SHRES_I2C5));
```

Comment:
I2C5 subsystem will no be affected by this change. I think `STM32MP1_SHRES_I2C5` should not be added as the underlying HW will always be accessible from the non secure world.

Removing definition and support of `STM32MP1_SHRES_I2C5`, I think the change you need in shared_resrouces.c is:
```patch
--- a/core/arch/arm/plat-stm32mp1/shared_resources.c
+++ b/core/arch/arm/plat-stm32mp1/shared_resources.c
@@ -319,6 +319,10 @@ static void register_periph_iomem(vaddr_t base, enum shres_state state)
        case UART7_BASE:
        case UART8_BASE:
        /* Fall through */
+#endif
+#ifdef CFG_WITH_NSEC_I2CS
+       case I2C5_BASE:
+       /* Fall through */
 #endif
        case IWDG2_BASE:
                /* Allow drivers to register some non-secure resources */